### PR TITLE
Refactors upload_cases

### LINKED ISSUE #65: test_run_external_algorithm tests too much
The test_run_external_algorithm tests that an algorithm has run, but only needs to test if a job has been scheduled.

### PR DESCRIPTION
**Pull request by jmsmkn**
This PR refactors the case upload helper functions, using slugs for consistency and removing duplicated functionality.

The tests will fail until https://github.com/comic/grand-challenge.org/pull/1626 is merged to master, and this should be released as soon as that PR is deployed.

Closes #65 

TODO for the future: we should check that the object that the user wants to upload to is correct before starting the upload session, check that the files exist, and the memory usage of the file uploads does not look great here either.

### CODE REVIEW

**Comment by jmsmkn** [comment]
@miriam-groeneveld comic/grand-challenge.org#1626 has been merged and the actions are now green after the outage yesterday, so this is finally ready for review :-)

File: README.rst [hunk 1]
```diff
@@ -29,7 +29,50 @@ Python client for the grand-challenge.org API
...
+
+First, you will need to authorise the client using your personal API token.
+
+.. code:: python
+
+    from gcapi import Client
+    
+    c = Client(token="Your personal API Token")
+
+Then, prepare the list of files you want to upload.
```
Comment (miriam-groeneveld) [inline README.rst:48]: It's one image right, although possibly multiple files? So maybe 
```suggestion
Then, prepare the list of files for each image you want to upload.
```

**Review by miriam-groeneveld — APPROVED** [review]
LGTM, I added some suggestions for the readme.

**Comment by jmsmkn** [comment]
Thanks! I will release a new version this morning.

**Pull request merged**